src/analysis/normed_space/operator_norm.lean

```diff
@@ -335,6 +336,18 @@ begin
     rwa [norm_inv, div_eq_mul_inv, inv_inv] }
 end
 
+/-- For a continuous real linear map `f`, if one controls the norm of every `f x`, `∥x∥ = 1`, then
+one controls the norm of `f`. -/
+lemma op_norm_le_of_unit_norm [normed_space ℝ E] [normed_space ℝ F] {f : E →L[ℝ] F} {C : ℝ}
+  (hC : 0 ≤ C) (hf : ∀ x, ∥x∥ = 1 → ∥f x∥ ≤ C) : ∥f∥ ≤ C :=
```

Comment:
I think that the file `normed_space.is_R_or_C` precisely exists for the purpose of having `is_R_or_C`-scalars-specific facts about normed spaces.  Can you move to there, generalize to scalars `is_R_or_C`, and rewrite to use the lemma `norm_smul_inv_norm` already existing there?  Here's a first version, you can probably golf it further :)

```lean
lemma op_norm_le_of_unit_norm  {f : E →L[𝕜] F} {C : ℝ}
  (hC : 0 ≤ C) (hf : ∀ x, ∥x∥ = 1 → ∥f x∥ ≤ C) : ∥f∥ ≤ C :=
begin
  refine f.op_norm_le_bound' hC (λ x hx, _),
  have H₁ : ∥((∥x∥⁻¹ : 𝕜) • x)∥ = 1 := norm_smul_inv_norm (by simpa using hx),
  have H₂ := hf _ H₁,
  rwa [map_smul, norm_smul, norm_inv, is_R_or_C.norm_coe_norm, ← div_eq_inv_mul, div_le_iff] at H₂,
  exact (norm_nonneg x).lt_of_ne' hx,
end
```

Same for the `nnnorm` version.

(I just noticed that actually, over in that file, there is a lemma `op_norm_bound_of_ball_bound` which looks like it should work for a nondiscrete normed field, but that's a discussion for another day ...)

Reply:
A normed space over an `is_R_or_C` is a normed space over reals, isn't it? Or it needs a `letI`?